src/main/java/io/fabric8/jenkins/openshiftsync/BuildSyncRunListener.java

```diff
@@ -201,9 +206,15 @@ private void upsertBuild(Run run, RunExt wfRunExt) {
       return;
     }
 
+    BluePipeline blueResource = (BluePipeline) BluePipelineFactory.resolve(run.getParent());
     String rootUrl = OpenShiftUtils.getJenkinsURL(getOpenShiftClient(), cause.getNamespace());
     String buildUrl = joinPaths(rootUrl, run.getUrl());
     String logsUrl = joinPaths(buildUrl, "/consoleText");
+    String logsConsoleUrl = joinPaths(buildUrl, "/console");
+    String logsBlueOceanUrl = null;
+    if (blueResource != null)
+        logsBlueOceanUrl = joinPaths(rootUrl, "/blue", "/organizations", "/" + blueResource.getOrganization(), "/" + run.getParent().getName(), "/detail",
+                "/" + run.getParent().getName(), "/" + run.getNumber(), "/pipeline");
```

Comment:
I was just mimicking the prior uses on the latter point.  But on further review, you are correct, `joinPaths` does seem to add separators.  I'll re-code and confirm.

On the first point, after a quick search there are enough jira's out there about encoding that I'll go ahead and add that too.  Certainly in this case we are constructing more of the URL vs. getting a URL from base Jenkins for the other cases.